src/plopp/core/limits.py

```diff
@@ -1,39 +1,64 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
 
-from typing import Literal, Tuple
+from typing import Literal, Tuple, Union
 
 import numpy as np
-from scipp import Variable, scalar
+from scipp import DataArray, Variable, scalar, stddevs
+
+from .utils import merge_masks
 
 
 def find_limits(
-    x: Variable, scale: Literal['linear', 'log'] = 'linear'
+    x: Union[Variable, DataArray], scale: Literal['linear', 'log'] = 'linear'
 ) -> Tuple[Variable, ...]:
     """
     Find sensible limits, depending on linear or log scale.
     If there are no finite values in the array, raise an error.
     If there are no positive values in the array, and the scale is log, fall back to
     some sensible default values.
     """
-    v = x.values
-    finite_inds = np.isfinite(v)
-    if np.sum(finite_inds) == 0:
+    m = np.ones_like(x.values, dtype='bool')
+    if isinstance(x, DataArray) and len(x.masks):
+        m &= ~merge_masks(x.masks).values
+    if not np.any(m):
+        m = np.ones_like(x.values, dtype='bool')
+
+    v = x.values[m]
+    if x.ndim == 1:
```

Comment:
I don't think you can rely on the data being 1d, because the data sent to a `scatter3d` plot will also be 1d, and you will get a 1d data from the `colormapper`. In that case you don't want to take into account the variances.